modules/tensor_mechanics/include/userobjects/CrackFrontDefinition.h

```diff
@@ -41,6 +41,12 @@ class CrackFrontDefinition : public GeneralUserObject, public BoundaryRestrictab
   virtual void finalize() override;
   virtual void execute() override;
 
+  /**
+   * Change the number of crack front nodes. As the crack grows, the number of crack fronts nodes
+   * may keep increasing in many cases.
+   */
+  void changeCrackFrontNodeNumber(const std::size_t num_points);
```

Comment:
I don't really like the way that the user is currently specifying the number of points that the points provider gives us in CrackFrontDefinition. I recall having a conversation with @jiangwen84 about this when he first added this, and we must have decided that the current approach was best, but this case where the cutter mesh is changing the number of points demonstrates why this is a problem.

I would prefer to have a method like `getNumberOfCrackPoints()` on `CrackFrontPointsProvider`, and have `CrackFrontDefinition` ask `CrackFrontPointsProvider` how many there are, rather than the other way around. For the `CrackFrontPointsProvider` objects like the circle cut that don't have a discrete set of points, maybe we could have an optional user-settable parameter. You wouldn't need to set it all the time, but if you tried to use the cutter with the fracture integrals, you'd get an error if it wasn't set.

Reply:
I left the Crack Front Points Management unchanged due to the difficulty described in a note below.  However, I did create a function getNumberOfCrackPoints() just for CrackMeshCut3DUserObject.  In the future when the Crack Front Points Management is sorted out, you can use it to override the virtual function to be defined in the Provider. 

> A problem with this is that calcNumCrackFrontPoints() in DomainIntegralAction needs the number of crack front points, and it is called before the CircleCutUserObject is ever constructed.